spec/calculate_spec.rb

```diff
@@ -17,58 +16,55 @@
 
       expect(Calculate.format_date).to eql(today_date_formatted)
     end
-    it 'if date passed in, return passed in value', :focus do
-      expect(Calculate.format_date("123456")).to eql("123456")
-    end
-    it 'if date passed in, raises error if length does not equal 6' do
-      expect{Calculate.format_date("31420")}.to raise_error(ArgumentError)
-    end
-    it 'if date passed in, raises error if contains characters besides numbers' do
-      expect{Calculate.format_date("453f34")}.to raise_error(ArgumentError)
-    end
+
+      context 'when date passed in' do
+
+        it 'return passed in value'  do
+          expect(Calculate.format_date("123456")).to eql("123456")
+        end
+
+        context 'when length does not equal 6' do
+          it 'raises error' do
+            expect{Calculate.format_date("31420")}.to raise_error(ArgumentError)
+          end
+        end
+
+        context 'when containing characters besides numbers' do
+          it 'raises error ' do
+            expect{Calculate.format_date("453f34")}.to raise_error(ArgumentError)
+          end
+        end
+      end
   end
 
   describe '::offsets' do
-
     it 'returns hash containing last four digits of date squared' do
-      formatted_date = '011519'
-
-      expect(Calculate.offsets(formatted_date)).to include({a_offset:7,b_offset:3,c_offset:6,d_offset:1 })
+      expect(Calculate.offsets('011519')).to include({a_offset:7,b_offset:3,c_offset:6,d_offset:1 })
     end
     it 'returns hash containing last four digits of date squared test 2' do
-      formatted_date = '011517'
-
-      expect(Calculate.offsets(formatted_date)).to include({a_offset:1,b_offset:2,c_offset:8,d_offset:9 })
+      expect(Calculate.offsets('011517')).to include({a_offset:1,b_offset:2,c_offset:8,d_offset:9 })
     end
-
   end
 
   describe '::keys' do
     it 'splits random 5 digit number into 4 two-digit hash entries' do
```

Comment:
Kind of small. . . but nothing about this function forces the keys to be 'random'.

It seems we have a nice general function that will work for all strings (regardless of how we got the string).

No reason to leak the overall design into this helper function.